src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -83,7 +83,13 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
         $dateTimeFormat = $context[self::FORMAT_KEY] ?? null;
         $timezone = $this->getTimezone($context);
 
-        if (null === $data || !\is_string($data) || '' === trim($data)) {
+        $data = match ($dateTimeFormat){
```

Comment:
```suggestion
        if (\is_int($data) || \is_float($data)) {
            $data = match ($dateTimeFormat) {
                'U' => (int) $data,
                'U.u' => sprintf('%.6F', $data),
            };
        }
```

I'd rather not be extra strict with float vs int, let's accept both in both cases, that might event help with x86 archs.

Reply:
@nicolas-grekas thank you, hmm, it should be 'U' => (string) $data, right ?

Reply:
should be `(string) (int) $data`!

Reply:
or better: `sprintf('%d', $data)`

Reply:
Fixed